pkg/common/peertracker/peertracker_test.go

```diff
@@ -1,208 +1,206 @@
-//go:build !windows
-// +build !windows
-
 package peertracker
 
 import (
 	"fmt"
 	"net"
 	"os"
 	"os/exec"
-	"path"
+	"path/filepath"
 	"runtime"
 	"strconv"
 	"syscall"
 	"testing"
 
 	"github.com/sirupsen/logrus"
 	logtest "github.com/sirupsen/logrus/hooks/test"
-	"github.com/spiffe/spire/test/spiretest"
-	"github.com/stretchr/testify/suite"
+	"github.com/stretchr/testify/require"
 )
 
-func TestPeerTrackerTestSuite(t *testing.T) {
-	suite.Run(t, new(PeerTrackerTestSuite))
-}
-
-type PeerTrackerTestSuite struct {
-	suite.Suite
-
+type peertrackerTest struct {
 	childPath string
-	ul        *Listener
-	unixAddr  *net.UnixAddr
+	listener  *Listener
+	addr      net.Addr
 	logHook   *logtest.Hook
 }
 
-func (p *PeerTrackerTestSuite) SetupTest() {
-	tempDir := spiretest.TempDir(p.T())
-
-	p.childPath = path.Join(tempDir, "child")
-	buildOutput, err := exec.Command("go", "build", "-o", p.childPath, "peertracker_test_child.go").CombinedOutput() //nolint: gosec // false positive
+func setupTest(t *testing.T) *peertrackerTest {
+	childPath := filepath.Join(t.TempDir(), "child.exe")
+	buildOutput, err := exec.Command("go", "build", "-o", childPath, childSource).CombinedOutput()
 	if err != nil {
-		p.T().Logf("build output:\n%v\n", string(buildOutput))
-		p.FailNow("failed to build test child")
-	}
-
-	p.unixAddr = &net.UnixAddr{
-		Net:  "unix",
-		Name: path.Join(tempDir, "test.sock"),
+		t.Logf("build output:\n%v\n", string(buildOutput))
+		require.FailNow(t, "failed to build test child")
 	}
 
 	log, hook := logtest.NewNullLogger()
-	p.logHook = hook
+	logHook := hook
```

Comment:
nit: seems like we can just change the name  of `hook` to `logHook` and avoid creating a new variable?